### PR TITLE
DEVX-1576 Updates dependency libs to CP 5.4.1

### PR DESCRIPTION
**Pull request by rspurgeon**
Also updates the README instructions as well as the default values in the Makefile

After a 0.3.1 version is published to Confluent-Hub, we can build docker images from 5.4.1 for kafka-connect-datagen.

Of note: CP Operator has not yet released a 5.4.1 patch for their images, so we cannot yet build and publish datagen images based on `cp-server-connect-operator` for > 5.4.0

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -31,11 +31,11 @@
 
     <properties>
         <connect-runtime-version>2.0.0</connect-runtime-version>
-        <confluent.version>5.4.0</confluent.version>
+        <confluent.version>5.4.1</confluent.version>
```
Comment (C0urante) [inline pom.xml]:
```suggestion
```

We can (and should) remove this line; it's [already defined](https://github.com/confluentinc/common/blob/f668cc36070d6fdcbde9b7c9854e84d302c38346/pom.xml#L50) in the common parent pom.

File: pom.xml [hunk 2]
```diff
@@ -31,11 +31,11 @@
...
         <confluent.avro.generator.version>0.3.0</confluent.avro.generator.version>
         <junit.version>4.12</junit.version>
         <avro.version>1.8.1</avro.version>
-        <licenses.version>5.4.0</licenses.version>
+        <licenses.version>5.4.1</licenses.version>
```
Comment (C0urante) [inline pom.xml]:
```suggestion
```

Similarly to the `${confluent.version}` property, we can [remove this one too](https://github.com/confluentinc/common/blob/f668cc36070d6fdcbde9b7c9854e84d302c38346/pom.xml#L66).

File: pom.xml (hunk 1 above)
Comment (rspurgeon) [reply pom.xml]: Thanks @C0urante , addressed

**Comment by rspurgeon** [comment]
> is my understanding correct that you'd like us to publish version 0.3.1 after this PR is merged?

Yes @C0urante if possible.  I'm assuming here we'd want to build a new patch version of this library after updating the dependent library, so I assumed we'd go to 0.3.1.    After a new version is built I can build docker images for our demos.

**Comment by C0urante** [comment]
Version [0.3.1](https://github.com/confluentinc/kafka-connect-datagen/tree/v0.3.1) released!

**Pull request merged**